### PR TITLE
Stac result export

### PR DESCRIPTION
**Pull request by joaherrerama**
The PR:

- Creates a STAC Catalog per User
- Creates the option inside the **export** and **metada** option PC (Process Chain) 
- Store the **Raster File** (format supported temporally)
- Create paths for the assets
-  Create the extension for Procession, Projection and Raster in STAC

### CODE REVIEW

File: src/actinia_core/core/stac_exporter_interface.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
+# -*- coding: utf-8 -*-
+#######
+# actinia-core - an open source REST API for scalable, distributed, high
+# performance processing of geographical data that uses GRASS GIS for
+# computational tasks. For details, see https://actinia.mundialis.de/
+#
+# Copyright (c) 2021 mundialis GmbH & Co. KG
```
Comment (mmacata) [inline src/actinia_core/core/stac_exporter_interface.py:7]:
```suggestion
# Copyright (c) 2021-2022 mundialis GmbH & Co. KG
```

File: src/actinia_core/core/stac_exporter_interface.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+"""
+This code shows the STACExporter model which help actinia core
+to export and manage the STAC items a global collection configuration
+
+"""
+
+# from PyQt4.QtCore import *
+# from PyQt4.QtXml import *
+# from config import global_config
```
Comment (mmacata) [inline src/actinia_core/core/stac_exporter_interface.py:32]: Is this needed or can it be removed?

File: src/actinia_core/core/stac_exporter_interface.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+__license__ = "GPLv3"
+__author__ = "Jorge Herrera"
+__copyright__ = "Copyright 2021, mundialis GmbH & Co. KG"
```
Comment (mmacata) [inline src/actinia_core/core/stac_exporter_interface.py:36]:
```suggestion
__copyright__ = "Copyright 2021-2022, mundialis GmbH & Co. KG"
```

File: src/actinia_core/core/stac_exporter_interface.py [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+    has_plugin = True
+except Exception:
+    has_plugin = False
+
+
+class STACExporter:
+
+    def stac_collection_initializer():
+        """
+        Initilize the STAC Catalog for the different outputs in actinia
```
Comment (mmacata) [inline src/actinia_core/core/stac_exporter_interface.py:66]:
```suggestion
        Initialize the STAC Catalog for the different outputs in actinia
```

File: src/actinia_core/core/stac_exporter_interface.py [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+                     output_type: str = None):
+        """
+        This function build the STAC ITEM and implement the following extension:
+            - Projection
+            - Raster
+        Parameter:
+            Input:
+                - output_path = Path to the source
+                - filename =  name of the source
+                - output_type =  type of object (raster, vector)
```
Comment (mmacata) [inline src/actinia_core/core/stac_exporter_interface.py:96]: Duplicate (`Parameter`, `Input`). In actinia the docs are often like:
```
        Args:
            user_id (str): The user id
            resource_id (str): The resource id
            iteration (int): The iteration of the job

        Returns:
            str: The generated DB resource id
```

File: src/actinia_core/core/stac_exporter_interface.py [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+
+            # Add item to Catalog
+            catalog.add_item(item)
+
+            # Update redis catalog
+            self._update_catalog_redis(catalog)
+
+        # TODO
+        elif output_type == "vector":
+            raise AsyncProcessTermination("Still under developments")
```
Comment (mmacata) [inline src/actinia_core/core/stac_exporter_interface.py:157]: To not create too high expectations:
```suggestion
            raise AsyncProcessTermination("Not implemented yet.")
```

File: tests/test_async_processing_export.py [hunk 7]
```diff
@@ -256,6 +256,44 @@
...
+process_chain_short_stac = {
+    1: {
+        "module": "g.region",
+        "inputs": {
+            "raster": "elevation@PERMANENT"
+        },
+        "flags": "p",
+        "verbose": True
+    },
+    2: {
```
Comment (mmacata) [inline tests/test_async_processing_export.py:268]: Just a hint that this is the old PC style and we want to replace it in the future. See https://github.com/mundialis/actinia_core/issues/245 (no need to adjust it in this PR)

**Review by mmacata — CHANGES_REQUESTED** [review]
Nice, thanks for this cool new feature!
I only have some smaller remarks, then please merge!

**Comment by mmacata** [comment]
It popped up again in my notifications and when I clicked it, it only showed the change from `2021` to `2021-2022`... so I approved and added the smiley. Only after I realized that there are more changes. But as my remarks were small anyway, I just left it the way it is ;)

**Comment by mmacata** [comment]
Ah - but I forgot about the API docs - please add them!

**Review by mmacata — APPROVED** [review]
Thanks! I only suggested some typo fixes, else please merge!

File: src/actinia_core/models/process_chain.py [hunk 8]
```diff
@@ -266,6 +266,34 @@ class OutputParameter(IOParameterBase):
...
+            },
+            'type': {
+                'type': 'string',
+                'description': 'The type of the output. In this case'
+                               'metadata is the only option currently supported',
+                'enum': ['metadata']
+            },
+            'output_layer': {
+                'type': 'string',
+                'description': 'Prefix for STAC item name and ID. REQUIRED'
```
Comment (mmacata) [inline src/actinia_core/models/process_chain.py:288]: Is it really required? I thought the previous comment was more clear in that it uses the output map name if not set - at least this is what I was expecting. Just the word `vector` didn't fit too well as all types of output can be used.

Comment (joaherrerama) [reply src/actinia_core/models/process_chain.py:288]: Yes, I think we defined it as the attribute to give the name to the `STRDS`

File: src/actinia_core/models/process_chain.py [hunk 9]
```diff
@@ -266,6 +266,35 @@ class OutputParameter(IOParameterBase):
...
+            },
+            'type': {
+                'type': 'string',
+                'description': 'The type of the output. In this case'
+                               'metadata is the only option currently supported',
+                'enum': ['metadata']
+            },
+            'output_layer': {
+                'type': 'string',
+                'description': 'Prefix for STAC item name and ID.If not specified,'
```
Comment (mmacata) [inline src/actinia_core/models/process_chain.py:288]:
```suggestion
                'description': 'Prefix for STAC item name and ID. If not specified,'
```

**Comment by mmacata** [comment]
Alright - then one last suggestion and then please merge :)

**Pull request merged**